pmd-java/src/main/resources/category/java/codestyle.xml

```diff
@@ -1809,6 +1809,33 @@ public class Foo {
         </example>
     </rule>
 
+
+    <rule name="UseDiamondOperator"
+          language="java"
+          message="Use diamond operator"
+          class="net.sourceforge.pmd.lang.rule.XPathRule"
+          minimumLanguageVersion="1.7">
```

Comment:
you should add the `since` and `externalInfoUrl` attributes.

The since should point to the PMD version that introduced the rule (6.11.0), and the external url should be `${pmd.website.baseurl}/pmd_rules_java_codestyle.html#usediamondoperator `